src/3d/qgs3dutils.cpp

```diff
@@ -442,6 +442,15 @@ QgsVector3D Qgs3DUtils::worldToMapCoordinates( const QgsVector3D &worldCoords, c
                       worldCoords.y() + origin.z() );
 }
 
+QgsVector3D Qgs3DUtils::transformCoordinates( const QgsVector3D &coords, QgsCoordinateTransform coordTrans )
+{
+  double x = coords.x();
+  double y = coords.y();
+  double z = coords.z();
+  coordTrans.transformInPlace( x, y, z );
```

Comment:
all calls to coordinate transformation need to be wrapped in a

```suggestion
  try
  {
     coordTrans.transformInPlace( x, y, z );
  }
  catch ( QgsCsException& )
  {
     ... ?
  }
```

block

Reply:
(usually the catch block just has a `QgsDebugMsg( QStringLiteral("Error transforming point coordinate")) ` type debug message)

Reply:
Calling QgsDebugMsg for every point might slowdown QGIS so I think I will be prining only once for every processNode call